src/main/kotlin/org/rust/cargo/project/settings/RustfmtProjectSettingsService.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.settings
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import org.jetbrains.annotations.TestOnly
+import org.rust.cargo.toolchain.RustChannel
+
+interface RustfmtProjectSettingsService {
```

Comment:
I have created a new project service for all rustfmt settings (including the old ones from `RustProjectSettingsService`). I'm not sure if this is the right idea, but if it's OK, I will add some code for the migration.

Reply:
I'm not sure how to move `useRustfmt` and `runRustfmtOnSave` values from `RustProjectSettingsServiceImpl ` to `RustfmtProjectSettingsServiceImpl` since they both load independently of each other.

Reply:
But you can force proper order if you call one service inside other one.
There is `com.intellij.openapi.components.PersistentStateComponent#noStateLoaded` that called only when there isn't persistent state for the service. It can be the first service creation after plugin update or if all settings are default and weren't saved to the disk last time.
We can use this method to copy previously saved settings from `RustProjectSettingsService`. Of course, it requires keeping necessary fields in `RustProjectSettingsService` for a while but I don't think it's a problem. Just mark these fields as deprecated and add `BACKCOMPAT: 2022.1` comment to drop it